README.md

```diff
@@ -1,16 +1,164 @@
+change what they inherited.
+
+If you need to modify a child's inherited properties or methods, why are we
+inheriting them in the first place?
+
+Below are two examples, one that violates Liskov's principle, and one that
+upholds it:
+
+#### Violates Substitution Principle:
+
+```js
+class Reptile {
+	constructor(name) {
+		this.name = name;
+	}
+	move() {
+		return `${this.name} crawls away`;
+	}
+}
+
+// Lizard inherits `move()` because it crawls
+class Lizard extends Reptile {}
+
+//  Snake overrides `move()` because it cannot crawl
+class Snake extends Reptile {
+	move() {
+		return `${this.name} slithers away`;
+	}
+}
+```
+
+The `Snake` class is a subtype of class `Reptile`, but overrides the `move()`
+method because the original doesn't apply. If we created an instance of
+`Reptile`:
+
+```js
+let tricky = new Reptile('Tricky');
+```
+
+and an instance of `Snake`:
+
+```js
+let basilisk = new Snake('Basilisk');
+```
+
+We see that `tricky` cannot be replaced with `basilisk` without changing
+behavior:
+
+```js
+tricky.move(); // => "Tricky crawls away"
+basilisk.move(); // => "Tricky slithers away"
+```
+
+#### Upholds Substitution Principle:
+
+So how do we stop violating Liskov's principle? We can either choose to not
+inherit from the same parent, _or_ we can create a grandparent class, one
+that shares all methods and properties with all children and grandchildren:
+
+```js
+class Reptile {
+	constructor(name) {
+		this.name = name;
+	}
+}
+
+class LeglessReptile extends Reptile {
+	move() {
+		return `${this.name} slithers away`;
+	}
+}
+
+class LeggedReptile extends Reptile {
+	move() {
+		return `${this.name} crawls away`;
+	}
+}
+
+class Lizard extends LeggedReptile {}
+class Snake extends LeglessReptile {}
+```
+
+Now we've got two levels of inheritance. `Reptile` sets up the `constructor`
+that all children and grandchildren inherit. Because the `move()` method
+needs to behave differently for legged and legless reptiles, it is defined
+differently in both `LeglessReptile` and `LeggedReptile`.
+
+With this structure, if an instance of `Lizard` replaced an instance of
+`LeggedReptile` _or_ `Reptile`, it will work correctly. The same goes for an
+instance of `Snake`.
+
+## Recognize the Benefits of Upholding Liskov's Substitution Principle
+
+Both of the above examples _work_. There is no syntax error if you choose to
+ignore Liskov's substitution principle. This is considered a purely _semantic_
+design choice.
+
+The benefit of following this principle is that no matter how complicated
+inheritance gets, you always assume that whatever a parent class has, a child
+class will have too. Looking at a parent class should give us some insight into
+the functionality of any children, grandchildren, great grandchildren, etc...
+
+Having chains of inheritance where children might fundamentally change the data
+and behaviors they inherit can introduce bugs, but more importantly, can
```

Comment:
This sentence is a little hard to comprehend. Maybe "If we have chains of inheritance where children might fundamentally change the data and behaviors they inherit, it can introduce bugs. More importantly, it can..."